File: Swift-C++-Testsuite/test/structsAndClasses/Swift/Classes/OverrideInit.swift
@main
struct MyMain {

    static func main() {
        let bicycle = Bicycle()
        let hoverboard = Hoverboard(color: "silver")
        let hovDes = hoverboard.description
        let biDes = bicycle.description
    }
}

class Vehicle {
    var numberOfWheels = 0
    var description: String {
        return "\(numberOfWheels) wheel(s)"
    }
}

class Bicycle: Vehicle {
    override init() {
        super.init()
        numberOfWheels = 2
    }
}

class Hoverboard: Vehicle {
    var color: String
    init(color: String) {
        self.color = color
        // super.init() implicitly called here
    }
    override var description: String {
        return "\(super.description) in a beautiful \(color)"
    }
}

File: Swift-C++-Testsuite/test/structsAndClasses/C++/Classes/OverrideInit.cpp
#include <string>



class Vehicle {
public:
	int numberOfWheels = 0;
	std::string description() {
		return std::to_string(numberOfWheels) + " wheel(s)";
	}
};

class Bicycle : public Vehicle {
public:
	Bicycle(): Vehicle() {
		numberOfWheels = 2;
	}
};

class Hoverboard : public Vehicle {
private:
	std::string color;
public:
	Hoverboard(std::string color) : color(color) {};
	std::string description() {
		return Vehicle::description() + " in a beautiful " + color;
	}
};

int main()
{
	Bicycle bicycle;
	Hoverboard hoverboard("silver");
	std::string hovDes = hoverboard.description();
	std::string biDes = bicycle.description();
	return 0;
}